src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -961,6 +961,65 @@ public override SyntaxTree SyntaxTree
         {
+            var bindingRoot = GetBindingRootOrInitializer(node);
+
+            var statementOrRootOperation = GetStatementOrRootOperation(bindingRoot, options, cancellationToken);
+            if (statementOrRootOperation == null)
+            {
+                return null;
+            }
+
+            // we might optimize it later
+            return statementOrRootOperation.DescendantsAndSelf().FirstOrDefault(o => !o.IsImplicit && o.Syntax == node);
+        }
+
+        private CSharpSyntaxNode GetBindingRootOrInitializer(CSharpSyntaxNode node)
+        {
+            var bindingRoot = GetBindingRoot(node);
+
+            // if binding root is parameter, make it equal value
+            var parameter = bindingRoot as ParameterSyntax;
+            if (parameter != null && parameter.Default?.FullSpan.Contains(node.Span) == true)
+            {
+                // we need to do this since node map doesn't contain bound node for parameter
+                bindingRoot = parameter.Default;
+            }
+
+            // if binding root is field variable declarator, make it initializer
+            var variableDeclarator = bindingRoot as VariableDeclaratorSyntax;
+            if (variableDeclarator != null && variableDeclarator.Initializer?.FullSpan.Contains(node.Span) == true)
+            {
+                if (variableDeclarator.Parent?.Parent.IsKind(SyntaxKind.FieldDeclaration) == true ||
+                    variableDeclarator.Parent?.Parent.IsKind(SyntaxKind.EventFieldDeclaration) == true)
+                {
+                    // we need to do this since node map doesn't contain bound node for field variable declarator
+                    bindingRoot = variableDeclarator.Initializer;
+                }
+            }
+
+            // if binding root is enum member decleration, make it equal value
+            var enumMember = bindingRoot as EnumMemberDeclarationSyntax;
+            if (enumMember != null && enumMember.EqualsValue?.FullSpan.Contains(node.Span) == true)
+            {
+                // we need to do this since node map doesn't contain bound node for enum member decl
+                bindingRoot = enumMember.EqualsValue;
+            }
+
+            // if binding root is property member decleration, make it equal value
+            var propertyMember = bindingRoot as PropertyDeclarationSyntax;
+            if (propertyMember != null && propertyMember.Initializer?.FullSpan.Contains(node.Span) == true)
+            {
+                // we need to do this since node map doesn't contain bound node for property initializer
+                bindingRoot = propertyMember.Initializer;
+            }
+
+            return bindingRoot;
+        }
+
+        private IOperation GetStatementOrRootOperation(CSharpSyntaxNode node, GetOperationOptions options, CancellationToken cancellationToken)
```

Comment:
>GetOperationOptions options [](start = 78, length = 27)

Do we still need to pass options and why?

Reply:
I just renamed existing method. I guess I can simplify existing one since there will be no high/low bound node for statement and root?